crates/precompile/src/secp256k1.rs

```diff
@@ -45,20 +45,23 @@ mod secp256k1 {
 #[allow(clippy::module_inception)]
 mod secp256k1 {
     use primitives::{alloy_primitives::B512, keccak256, B256};
-    use secp256k1::{
+    use rust_secp256k1::{
         ecdsa::{RecoverableSignature, RecoveryId},
-        Message, SECP256K1,
+        Message, Secp256k1,
     };
 
     // Silence the unused crate dependency warning.
     use k256 as _;
+    use secp256k1 as _;
+
+    pub fn ecrecover(sig: &B512, recid: u8, msg: &B256) -> Result<B256, rust_secp256k1::Error> {
+        let engine = Secp256k1::new();
```

Comment:
does `rust_secp256k1` have something as a static context? for `secp256k1` this was one of the performance degradation and that is why we use `SECP256K1.recover_ecdsa` as it is faster